src/Host/Client/Impl/Host/RHost.cs

```diff
@@ -275,7 +288,15 @@ public sealed partial class RHost : IDisposable, IRExpressionEvaluator {
             } else {
                 result = new REvaluationResult(response[2], error, parseStatus);
             }
-            request.CompletionSource.SetResult(result);
+            request.CompletionSource.TrySetResult(result);
+        }
+
+        private void CancelRemainingEvaluationRequests() {
+            var requests = _evalRequests.Values;
+            _evalRequests.Clear();
```

Comment:
I assume this sequence does not have to be atomic (i.e. between Clear() and between setting cancellations there may be calls using values in requests)